### ISSUE TITLE
`opam publish` fails with `HTTP Error 404 Not Found`

### ISSUE THREAD

**Issue by XVilka** [ocaml-opam/opam-publish#104]
```
[i] ℤ opam publish                                                                                                                                                                                                                 13:26:39 

The following will be published:
  - radare2 version 0.0.4 with opam file at /home/akochkov/radare/ocaml-radare2/radare2.opam
    archive at https://github.com/fxfactorial/ocaml-radare2/archive/v0.0.4.tar.gz


Continue ? You will be shown the patch before submitting. [Y/n] y
Fetching the package repository, this may take a while...

File a pull-request for this patch ? [Y/n] Y
HTTP Error 404 Not Found
Headers:
access-control-allow-origin: *
access-control-expose-headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
content-length: 106
content-security-policy: default-src 'none'
content-type: application/json; charset=utf-8
date: Tue, 26 May 2020 05:39:06 GMT
referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
server: GitHub.com
status: 404 Not Found
strict-transport-security: max-age=31536000; includeSubdomains; preload
vary: Accept-Encoding, Accept, X-Requested-With
x-content-type-options: nosniff
x-frame-options: deny
x-github-media-type: github.v3; format=json
x-github-request-id: A2FE:1590:323810:3EFE5D:5ECCAB79
x-ratelimit-limit: 60
x-ratelimit-remaining: 59
x-ratelimit-reset: 1590475145
x-xss-protection: 1; mode=block

Body:
{"message":"Not Found","documentation_url":"https://developer.github.com/v3/pulls/#create-a-pull-request"}
[ERROR] HTTP Error 404 Not Found
        Headers:
        access-control-allow-origin: *
        access-control-expose-headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type,
        Deprecation, Sunset
        content-length: 106
        content-security-policy: default-src 'none'
        content-type: application/json; charset=utf-8
        date: Tue, 26 May 2020 05:39:06 GMT
        referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
        server: GitHub.com
        status: 404 Not Found
        strict-transport-security: max-age=31536000; includeSubdomains; preload
        vary: Accept-Encoding, Accept, X-Requested-With
        x-content-type-options: nosniff
        x-frame-options: deny
        x-github-media-type: github.v3; format=json
        x-github-request-id: A2FE:1590:323810:3EFE5D:5ECCAB79
        x-ratelimit-limit: 60
        x-ratelimit-remaining: 59
        x-ratelimit-reset: 1590475145
        x-xss-protection: 1; mode=block

        Body:
        {"message":"Not Found","documentation_url":"https://developer.github.com/v3/pulls/#create-a-pull-request"}
```

**Comment by kit-ty-kate** [comment]
I got the same issue.

**Comment by kit-ty-kate** [comment]
For the record, the issue disappeared when I downgraded the `github` package to `4.2.0`. My `opam-publish` installation wasn't up-to-date and lacked this fix from opam-repository: https://github.com/ocaml/opam-repository/pull/16262

**Comment by Aaylor** [comment]
Hello,

You can take a look at my comment made on #97, the issue should be fixed with the new upcoming version `4.3.2`. Feel free to open a new issue on `ocaml-github`, or to ping me directly, if you face another issue with `ocaml-github`.
Sorry, for the inconvenience!

**Comment by charlesetc** [comment]
Hello, I got a similar error trying to publish to opam... this seems like a decent place to ask but let me know if you'd rather me make an issue somewhere else. Here's the error:
```
[ERROR] HTTP Error 404 Not Found
        Headers:
        access-control-allow-origin: *
        access-control-expose-headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining,
        X-RateLimit-Used, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type,
        Deprecation, Sunset
        content-length: 74
        content-security-policy: default-src 'none'
        content-type: application/json; charset=utf-8
        date: Tue, 08 Dec 2020 04:59:55 GMT
        referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
        server: GitHub.com
        status: 404 Not Found
        strict-transport-security: max-age=31536000; includeSubdomains; preload
        vary: Accept-Encoding, Accept, X-Requested-With
        x-accepted-oauth-scopes: repo
        x-content-type-options: nosniff
        x-frame-options: deny
        x-github-media-type: github.v3; format=json
        x-github-request-id: A0B0:63DD:361B0A9:5BD73FF:5FCF084B
        x-oauth-scopes: repo
        x-ratelimit-limit: 5000
        x-ratelimit-remaining: 4995
        x-ratelimit-reset: 1607406751
        x-ratelimit-used: 5
        x-xss-protection: 1; mode=block

        Body:
        {"message":"Not Found","documentation_url":"https://docs.github.com/rest"}
```

The command I ran was `https://hg.sr.ht/~etc/feather/archive/0.1.0.tar.gz`, filled in my username and access token, and then got this error. I tried the github `4.2.0` package, the github `4.3.2` package, and building from the master branch. All with opam-publish `2.0.2.` Any insights are much appreciated, thanks!

**Comment by rjbou** [comment]
ping @Aaylor, an idea ?

**Comment by Aaylor** [comment]
Sorry for the delay!

I've given an answer in the other issue #119.

**Comment by rjbou** [comment]
Fixed by #121
